src/internal/packager/helm/zarf.go

```diff
@@ -61,20 +62,20 @@ func (h *Helm) UpdateZarfRegistryValues(ctx context.Context) error {
 	}
 	waitCtx, waitCancel := context.WithTimeout(ctx, 60*time.Second)
 	defer waitCancel()
-	err = healthchecks.WaitForReady(waitCtx, h.cluster.Watcher, objs)
+	err = healthchecks.WaitForReady(waitCtx, opts.Cluster.Watcher, objs)
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
 // UpdateZarfAgentValues updates the Zarf agent deployment with the new state values
-func (h *Helm) UpdateZarfAgentValues(ctx context.Context) error {
+func UpdateZarfAgentValues(ctx context.Context, opts InstallUpgradeOpts) error {
 	l := logger.From(ctx)
 	spinner := message.NewProgressSpinner("Gathering information to update Zarf Agent TLS")
 	defer spinner.Stop()
 
-	deployment, err := h.cluster.Clientset.AppsV1().Deployments(cluster.ZarfNamespaceName).Get(ctx, "agent-hook", metav1.GetOptions{})
+	deployment, err := opts.Cluster.Clientset.AppsV1().Deployments(cluster.ZarfNamespaceName).Get(ctx, "agent-hook", metav1.GetOptions{})
```

Comment:
FUTURE: This is a lot of .'s and mutation but not something we need to fix right away.